tests/wandb_run_test.py

```diff
@@ -300,3 +300,42 @@ def test_unlogged_artifact_in_config(live_mock_server, test_settings):
             str(e_info.value)
             == "Cannot json encode artifact before it has been logged or in offline mode."
         )
+
+
+def test_anonymous_mode(live_mock_server, test_settings, capsys):
+    api_key = os.environ.pop("WANDB_API_KEY")
+    uname = os.environ.pop("WANDB_USERNAME")
+    test_settings.update({"anonymous": "must", "api_key": None})
+
+    run = wandb.init(settings=test_settings, anonymous="must")
```

Comment:
If for some reason there's an exception thrown in `wandb.init`, does that mean that we might not set `WANDB_USERNAME` and `WANDB_API_KEY` back to the original values? That could cause the other tests to fail unexpectedly.

Ideally, we would have something like:

```
with new_env(lambda env: # modify env as needed):
  run = wandb.init(...)
```

So that even if there's an error, the `with` guarantees that we restore the original env.

Reply:
you can mock/patch environ variables:
https://adamj.eu/tech/2020/10/13/how-to-mock-environment-variables-with-pytest/
https://docs.pytest.org/en/stable/monkeypatch.html#monkeypatching-environment-variables